src/Infrastructure/Identity/TokenService.cs

```diff
@@ -23,23 +25,46 @@ internal class TokenService : ITokenService
     private readonly MailSettings _mailSettings;
     private readonly JwtSettings _jwtSettings;
     private readonly FSHTenantInfo? _currentTenant;
+    private readonly ICaptchaService _capchaService;
+    private readonly IConfiguration _config;
 
     public TokenService(
         UserManager<ApplicationUser> userManager,
         IOptions<JwtSettings> jwtSettings,
         IStringLocalizer<TokenService> localizer,
         IOptions<MailSettings> mailSettings,
-        FSHTenantInfo? currentTenant)
+        FSHTenantInfo? currentTenant,
+        ICaptchaService captchaService,
+        IConfiguration configuration)
     {
         _userManager = userManager;
         _localizer = localizer;
         _mailSettings = mailSettings.Value;
         _jwtSettings = jwtSettings.Value;
         _currentTenant = currentTenant;
+        _capchaService = captchaService;
+        _config = configuration;
     }
 
     public async Task<TokenResponse> GetTokenAsync(TokenRequest request, string ipAddress, CancellationToken cancellationToken)
     {
+        var settings = _config.GetSection(nameof(CaptchaSettings)).Get<CaptchaSettings>();
```

Comment:
Not sure about the whole ICaptchaService api... 

It's not very user friendly to use...  

It should ultimately only be a single call here in this method...
_captchaSerice.CheckCaptcha(...) (the name is just a suggestion)

That method should do everything. Check the settings if it actually needs to do a check and then do the actual check against the parameters (why is that split into initialize and then execute anyway... seems very unintuitive?).

Reply:
_captchaSerice.ValildateCaptchaToken(...) totally agree with you, the whole implementation issue should be abstract